pex/interpreter_constraints.py

```diff
@@ -46,53 +48,58 @@ def create_message(self, preamble=None):
         """
         preamble = "{}\n\n".format(preamble) if preamble else ""
         if not self.candidates:
+            if self.failures:
+                return (
+                    "{preamble}"
+                    "Interpreters were found but they all appear to be broken:\n  {failures}"
```

Comment:
This should probably link to https://github.com/pantsbuild/pex/issues/1027 as well.

Reply:
Fixed.

The full matrix is now:
1. None
    ```
    Could not find a compatible interpreter.

    No interpreters could be found on the system.
    ```

2. None - All broken:
    ```
    Could not find a compatible interpreter.

    Interpreters were found but they all appear to be broken:
    1.) /usr/bin/python3.8:
    Traceback (most recent call last):
      File "<string>", line 5, in <module>
      File "/workspace/pex/interpreter.py", line 24, in <module>
        from pex.third_party.packaging import markers, tags
      File "/workspace/pex/vendor/_vendored/packaging/packaging/tags.py", line 7, in <module>
        import distutils.util
    ModuleNotFoundError: No module named 'distutils.util'

    (See https://github.com/pantsbuild/pex/issues/1027 for a list of known breaks and workarounds.)
    ```

3. Some but some broken:
    ```
    Could not find a compatible interpreter.

    Examined the following working interpreters:
    1.) /usr/bin/python2.7 CPython==2.7.18

    Skipped the following broken interpreters:
    1.) /usr/bin/python3.8:
    Traceback (most recent call last):
      File "<string>", line 5, in <module>
      File "/workspace/pex/interpreter.py", line 24, in <module>
        from pex.third_party.packaging import markers, tags
      File "/workspace/pex/vendor/_vendored/packaging/packaging/tags.py", line 7, in <module>
        import distutils.util
    ModuleNotFoundError: No module named 'distutils.util'

    (See https://github.com/pantsbuild/pex/issues/1027 for a list of known breaks and workarounds.)

    No working interpreter compatible with the requested constraints was found:
      CPython>=3.7
    ```

4. Some and none broken:
    ```
    Could not find a compatible interpreter.

    Examined the following interpreters:
    1.) /usr/bin/python2.7 CPython==2.7.18
    2.) /usr/bin/python3.8 CPython==3.8.2

    No interpreter compatible with the requested constraints was found:
      CPython>=3.9
    ```

Reply:
Excellent!